### PR TITLE
Tomas/add identity db

### LINKED ISSUE #14: Set up identity storage in database
Once the database is set up, tables relevant to user identity handling (e.g. user data and something related to authorization) should be added so that work can proceed on task [#6 (Add specific user page)](https://github.com/users/BenasB/projects/2/views/1?pane=issue&itemId=61722541)

### PR DESCRIPTION
**Pull request by TomasKvaraciejus**
* Added Users table to Identity project
* Added sample interaction with Identity project DataAccess layer 
* Minor changes to Docker configuration to correctly configure which ports are used for each project's api and db

Closes #14

### CODE REVIEW

File: Identity/Identity.API/Program.cs [hunk 1]
```diff
@@ -13,17 +18,32 @@
...
-        new()
-        {
-            Username = "johndoe"
-        }
+        u.Id,
+        u.Username,
+        u.PasswordHash,
+        u.Email,
+        u.Roles
+    });
```
Comment (6ar8nas) [inline Identity/Identity.API/Program.cs:30]: this shows warning for missing await operator. You can use .ToListAsync() here

File: Identity/Dockerfile [hunk 2]
```diff
@@ -10,7 +11,7 @@ COPY ["Identity.API/Identity.API.csproj", "Identity.API/"]
 COPY ["Identity.Public/Identity.Public.csproj", "Identity.Public/"]
```
Comment (6ar8nas) [inline Identity/Dockerfile:11]: Copy DataAccess project

File: Identity/Identity.DataAccess/Models/Entities/UserEntity.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+namespace Identity.API.Models.Entities;
+
+public class UserEntity
+{
+	[Key]
+	[DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+	public int Id { get; set; }
+
+	[Required]
+	public string Username { get; set; }
```
Comment (6ar8nas) [inline Identity/Identity.DataAccess/Models/Entities/UserEntity.cs:13]: There are quite a few warnings here as well. It seems like in Recipes DB pull request Rytis used required keyword rather than attribute.

File: Recipes/Recipes.API/Models/Entities/IngredientEntity.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+using System.ComponentModel.DataAnnotations;
+
+namespace Recipes.API.Models.Entities;
+public class IngredientEntity
+{
+    [Key]
+    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+    public int Id { get; set; }
+
+    public string Name { get; set; }
```
Comment (6ar8nas) [inline Recipes/Recipes.API/Models/Entities/IngredientEntity.cs:11]: This seems wrong. Why is this added here? We have previously decided on a different entity without RecipeIngredient relationship, which is already implemented.

**Review by 6ar8nas — CHANGES_REQUESTED** [review]

File: Identity/Dockerfile [hunk 5]
```diff
@@ -1,7 +1,8 @@
 ﻿FROM mcr.microsoft.com/dotnet/aspnet:8.0 AS base
 USER $APP_UID
 WORKDIR /app
-EXPOSE 8080
+EXPOSE 8081
+ENV ASPNETCORE_URLS=http://*:8081
```
Comment (BenasB) [inline Identity/Dockerfile:5]: Why is this needed? You can just specify a port binding when running a container, e.g. if you want to reach the container on port 8081 and the web server inside the container listens on 8080, you just specify `-p 8081:8080`, [docs](https://docs.docker.com/network/#published-ports)

File: Identity/Identity.API/Identity.API.csproj [hunk 6]
```diff
@@ -9,11 +9,16 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="8.0.1"/>
-        <PackageReference Include="Swashbuckle.AspNetCore" Version="6.4.0"/>
+        <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="8.0.1" />
+        <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="8.0.4">
+          <PrivateAssets>all</PrivateAssets>
+          <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+        </PackageReference>
```
Comment (BenasB) [inline Identity/Identity.API/Identity.API.csproj:16]: Is this needed on the `.API` project?

File: Identity/Identity.DataAccess/Models/Entities/UserEntity.cs (hunk 3 above)
Comment (BenasB) [reply Identity/Identity.DataAccess/Models/Entities/UserEntity.cs:13]: The required attribute makes it `NOT NULL` in SQL. You should also probably specify `public required type` as well

File: Recipes/Recipes.API/Models/Entities/IngredientEntity.cs (hunk 4 above)
Comment (BenasB) [reply Recipes/Recipes.API/Models/Entities/IngredientEntity.cs:11]: The commits seemed to be messed up. No changes should be done in `Recipes/`

**Review by BenasB — CHANGES_REQUESTED** [review]
I can't see a migration for the initialization of `UserEntity`

File: Identity/Dockerfile (hunk 5 above)
Comment (BenasB) [reply Identity/Dockerfile:5]: you should do that as well [here](https://github.com/BenasB/vu-psk/blob/54955f13b026c821fe8de45f3299a3ee6236f9c7/Identity/docker-compose.yaml#L7)

**Comment by 6ar8nas** [comment]
FIY you can check #24 changes. It addresses some of the issues that this seems to have as well.

File: Identity/Dockerfile (hunk 2 above)
Comment (TomasKvaraciejus) [reply Identity/Dockerfile:10]: Added Copy statement

File: Identity/Identity.API/Program.cs (hunk 1 above)
Comment (TomasKvaraciejus) [reply Identity/Identity.API/Program.cs]: Added .ToListAsync()

File: Identity/Identity.DataAccess/Models/Entities/UserEntity.cs (hunk 3 above)
Comment (TomasKvaraciejus) [reply Identity/Identity.DataAccess/Models/Entities/UserEntity.cs]: Added required keyword

File: Recipes/Recipes.API/Models/Entities/IngredientEntity.cs (hunk 4 above)
Comment (TomasKvaraciejus) [reply Recipes/Recipes.API/Models/Entities/IngredientEntity.cs]: Fixed commit issues

File: Identity/Identity.API/Identity.API.csproj (hunk 6 above)
Comment (TomasKvaraciejus) [reply Identity/Identity.API/Identity.API.csproj]: Nope, removed it

File: Identity/Identity.DataAccess/Models/Entities/UserEntity.cs [hunk 7]
```diff
@@ -10,16 +10,16 @@ public class UserEntity
 	public int Id { get; set; }
 
 	[Required]
-	public string Username { get; set; }
+	public required string Username { get; set; }
 
-	[Required]
-	public string PasswordHash { get; set; }
+    [Required]
```
Comment (6ar8nas) [inline Identity/Identity.DataAccess/Models/Entities/UserEntity.cs:15]: Some weird formatting mismatch happened

File: Identity/Dockerfile (hunk 5 above)
Comment (TomasKvaraciejus) [reply Identity/Dockerfile]: Changed this

File: Identity/Identity.DataAccess/Models/Entities/UserEntity.cs (hunk 7 above)
Comment (TomasKvaraciejus) [reply Identity/Identity.DataAccess/Models/Entities/UserEntity.cs:15]: Strange, it shows up on mobile but is not present in VS or GitHub on desktop?
image

**Comment by TomasKvaraciejus** [comment]
> A couple of small changes left

Fixed issues you pointed out. I also added EF Design package to API since it's necessary to perform migrations.

File: Identity/Identity.DataAccess/Models/Entities/UserEntity.cs (hunk 7 above)
Comment (6ar8nas) [reply Identity/Identity.DataAccess/Models/Entities/UserEntity.cs:15]: Got it. Perhaps someone else will format it with their changes if that shows up in their IDE.
